deploy/example.yaml

```diff
@@ -0,0 +1,207 @@
+    # or not, if set to false, this component will not be installed.
+    #
+    # Defaults to true.
+    enabled: "true"
+    # imageTag is the tag for the container image which will be used
+    # by this component.
+    #
+    # If not set, it will be the spec.version of the OpenEBS set
+    # above.
+    #
+    # +optional
+    imageTag:
+    # replicas specifies the number of replicas to be deployed.
+    #
+    # Defaults to 1.
+    replicas: 1
+    # sparse specifies the config for sparse pools i.e., whether it
+    # should be created by default or not when OpenEBS gets
+    # installed.
+    sparse:
+      # enabled decides whether default cstor
+      # sparse pool should be configured as a part of openebs
+      # installation.
+      # If "true" a default cstor sparse pool will be configured, if
+      # "false" it will not be configured.
+      # This value takes effect only if createDefaultStorageConfig is
+      # set to true.
+      #
+      # Defaults to false
+      enabled: "false"
+    # nodeSelector specifies on which nodes the pods of this
+    # component can be scheduled.
+    #
+    # Default to nil i.e., all the nodes present in the cluster
+    # i.e., the pods can be scheduled anywhere in the cluster.
+    nodeSelector:
+    # tolerations can be used if we want the pods of this component
+    # to tolerate some of the taints of nodes.
+    #
+    # Default to nil
+    tolerations:
+
+  # provisioner stores the configuration for OpenEBS provisioner
+  provisioner:
+    enabled: "true"
+    imageTag:
+    replicas: 1
+    nodeSelector:
+    tolerations:
+
+  # localProvisioner stores the configuration for OpenEBS local
+  # provisioner
+  localProvisioner:
+    enabled: "true"
+    imageTag:
+    replicas: 1
+    nodeSelector:
+    tolerations:
+
+  # snapshotOperator stores the configuration for snapshot operator
+  snapshotOperator:
+    enabled: "true"
+    controller:
+      imageTag:
+    provisioner:
+      imageTag:
+    replicas: 1
+    nodeSelector:
+    tolerations:
+
+  # ndm stores the configuration for node-disk-manager
+  ndm:
+    enabled: "true"
+    imageTag: "v0.4.5"
+    replicas: 1
+    sparse:
+      path:
+      size:
+      count:
+    # filters contain the config for excluding or including vendors,
+    # paths, etc.
+    #
+    # Defaults for excluding vendors will be CLOUDBYT,OpenEBS, paths
+    # will be loop,fd0,sr0,/dev/ram,/dev/dm-,/dev/md
+    filters:
+      OSDisk:
+        enabled:
+        exclude: "/,/etc/hosts,/boot"
+      vendor:
+        enabled:
+        exclude: "CLOUDBYT,OpenEBS"
+      path:
+        enabled:
+        exclude: "loop,fd0,sr0,/dev/ram,/dev/dm-,/dev/md"
+        include: ""
+    # probes can be used to configure ndm probes such as Seachest,
+    # smart, capacity, etc.
+    probes:
+      enable:
+      - seachest
+    nodeSelector:
+    tolerations:
+
+  # ndmOperator stores the configuration for ndm operator
```

Comment:
Next PR: Let us elaborate exactly the role of ndmOperator
Same comment for all components.